src/extension.js

```diff
@@ -77,6 +85,24 @@ function setGlobalCommandInstance(commandInstance) {
     globalCommand = commandInstance;
 }
 
+async function runWithCoverage(options) {
+    let command;
+
+    if (vscode.workspace.getConfiguration("better-phpunit").get("ssh.enable")) {
+        command = new RemotePhpUnitCommand( options );
+    } else if (vscode.workspace.getConfiguration("better-phpunit").get("docker.enable")) {
+        command = new DockerPhpUnitCommand( options );
+    } else {
+        command = new PhpUnitCommand( options );
+    }
```

Comment:
This "runWithCoverage" method seems redundant - maybe there is a better abstraction that doesn't duplicate this entire conditional block.

Reply:
I am not sure how you would like this to be. If I would use something like `vscode.commands.executeCommand`, to execute the default `run` and `run-suite` commands, from what I understand I must change the way they set the options and to get parameters as well. Unless there is an other way to alter the `command` variable.